### PR TITLE
Synchronize ORY/Hydra charts to latest version

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Update `hydra` and `hydra-maester` charts with the version `v0.5.5` from ORY.
Changes proposed in this pull request:

- Update `hydra` and `hydra-maester` charts

**Related issue(s)**
See: #9020

### CODE REVIEW

File: resources/ory/charts/hydra/templates/hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{{- if .Values.deployment.autoscaling.enabled }}
+apiVersion: autoscaling/v2beta2
+kind: HorizontalPodAutoscaler
+metadata:
```
Comment (colunira) [inline resources/ory/charts/hydra/templates/hpa.yaml:4]: We have a HPA for both oathkeeper and hydra in `resources/ory/templates/hpa.yaml`. We should think if we would like to remove it.

File: resources/ory/charts/hydra/values.yaml [hunk 2]
```diff
@@ -16,6 +18,10 @@ nameOverride: ""
 # Full chart name override
 fullnameOverride: ""
 
+## Pod priority
+## https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
+# priorityClassName: ""
+
```
Comment (colunira) [inline resources/ory/charts/hydra/values.yaml:22]: Why this is commented out?

**Review by colunira — COMMENTED** [review]
We should verify if secret migration when changing database works

File: resources/ory/charts/hydra/templates/deployment.yaml [hunk 3]
```diff
@@ -2,9 +2,18 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "hydra.fullname" . }}
+  {{- if .Release.Namespace }}
```
Comment (strekm) [inline resources/ory/charts/hydra/templates/deployment.yaml:5]: why do we need that if statement here? is there a possibility that we won't have value of Release.Namespace?

File: resources/ory/charts/hydra/templates/deployment.yaml [hunk 4]
```diff
@@ -157,18 +203,18 @@ spec:
...
           resources:
             {{- toYaml .Values.deployment.resources | nindent 12 }}
+      {{- if .Values.priorityClassName }}
+      priorityClassName: {{ .Values.priorityClassName }}
+      {{- end }}
       {{- with .Values.deployment.nodeSelector }}
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-    {{- if .Values.global.priorityClassName }}
```
Comment (strekm) [inline resources/ory/charts/hydra/templates/deployment.yaml:164]: shouldn't we use `global.priorityClassName` instead?

File: resources/ory/charts/hydra/charts/hydra-maester/values.yaml [hunk 5]
```diff
@@ -50,6 +61,11 @@ deployment:
   # Configure node tolerations.
   tolerations: []
   annotations: {}
+  podLabels: {}
+
+#TODO: Remove - job is removed
```
Comment (strekm) [inline resources/ory/charts/hydra/charts/hydra-maester/values.yaml:66]: i don't understand what is going on here o.o

File: resources/ory/charts/hydra/values.yaml (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply resources/ory/charts/hydra/values.yaml:22]: It's copied from ORY charts.

File: resources/ory/charts/hydra/templates/deployment.yaml (hunk 3 above)
Comment (Tomasz-Smelcerz-SAP) [reply resources/ory/charts/hydra/templates/deployment.yaml:5]: I don't think so. It's a copy from ORY chart, introduced by [this PR](https://github.com/ory/k8s/pull/82/files)

File: resources/ory/charts/hydra/templates/deployment.yaml (hunk 4 above)
Comment (Tomasz-Smelcerz-SAP) [reply resources/ory/charts/hydra/templates/deployment.yaml:164]: Good catch, fixed

File: resources/ory/charts/hydra/values.yaml (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply resources/ory/charts/hydra/values.yaml]: It was a copy from ORY charts, I've removed it.

File: resources/ory/charts/hydra/charts/hydra-maester/values.yaml (hunk 5 above)
Comment (Tomasz-Smelcerz-SAP) [reply resources/ory/charts/hydra/charts/hydra-maester/values.yaml:62]: One leftover from sync-ing :) Removed.

**Review by colunira — APPROVED** [review]
/meow

**Review by strekm — DISMISSED** [review]
/meow

**Review by strekm — APPROVED** [review]
/meow

**Pull request merged**